src/libsync/syncengine.cpp

```diff
@@ -365,395 +286,101 @@ void SyncEngine::conflictRecordMaintenance()
     }
+    if (item->_instruction == CSYNC_INSTRUCTION_UPDATE_METADATA && !item->isDirectory()) {
+        // For directories, metadata-only updates will be done after all their files are propagated.
 
-    switch (file->error_status) {
-    case CSYNC_STATUS_OK:
-        break;
-    case CSYNC_STATUS_INDIVIDUAL_IS_SYMLINK:
-        item->_errorString = tr("Symbolic links are not supported in syncing.");
-        break;
-    case CSYNC_STATUS_INDIVIDUAL_IGNORE_LIST:
-        item->_errorString = tr("File is listed on the ignore list.");
-        break;
-    case CSYNC_STATUS_INDIVIDUAL_IS_INVALID_CHARS:
-        if (item->_file.endsWith('.')) {
-            item->_errorString = tr("File names ending with a period are not supported on this file system.");
-        } else {
-            char invalid = '\0';
-            foreach (char x, QByteArray("\\:?*\"<>|")) {
-                if (item->_file.contains(x)) {
-                    invalid = x;
-                    break;
-                }
-            }
-            if (invalid) {
-                item->_errorString = tr("File names containing the character '%1' are not supported on this file system.")
-                                         .arg(QLatin1Char(invalid));
-            } else {
-                item->_errorString = tr("The file name is a reserved name on this file system.");
+        // Update the database now already:  New remote fileid or Etag or RemotePerm
+        // Or for files that were detected as "resolved conflict".
+        // Or a local inode/mtime change
+
+        // In case of "resolved conflict": there should have been a conflict because they
+        // both were new, or both had their local mtime or remote etag modified, but the
+        // size and mtime is the same on the server.  This typically happens when the
+        // database is removed. Nothing will be done for those files, but we still need
+        // to update the database.
+
+        // This metadata update *could* be a propagation job of its own, but since it's
+        // quick to do and we don't want to create a potentially large number of
+        // mini-jobs later on, we just update metadata right now.
+
+        if (item->_direction == SyncFileItem::Down) {
+            QString filePath = _localPath + item->_file;
+
+            // If the 'W' remote permission changed, update the local filesystem
+            SyncJournalFileRecord prev;
+            if (_journal->getFileRecord(item->_file, &prev)
+                && prev.isValid()
+                && prev._remotePerm.hasPermission(RemotePermissions::CanWrite) != item->_remotePerm.hasPermission(RemotePermissions::CanWrite)) {
+                const bool isReadOnly = !item->_remotePerm.isNull() && !item->_remotePerm.hasPermission(RemotePermissions::CanWrite);
+                FileSystem::setFileReadOnlyWeak(filePath, isReadOnly);
             }
+
+            _journal->setFileRecordMetadata(item->toSyncJournalFileRecordWithInode(filePath));
```

Comment:
There used to be code like this:
```
                if (other && other->type != ItemTypeVirtualFile && other->type != ItemTypeVirtualFileDownload) {
                    // Even if the mtime is different on the server, we always want to keep the mtime from
                    // the file system in the DB, this is to avoid spurious upload on the next sync
                    item->_modtime = other->modtime;
                    // same for the size
                    item->_size = other->size;
                }
```
in `treewalkFile`. But there's a good chance the way you're populating item now takes care of it.

Yes, I think the mtime/size changes to virtual files belong here. But not in this PR.